modules/azure/resourcegroup.go

```diff
@@ -0,0 +1,50 @@
+package azure
+
+import (
+	"context"
+	"testing"
+
+	"github.com/Azure/azure-sdk-for-go/services/resources/mgmt/2019-05-01/resources"
+	"github.com/stretchr/testify/require"
+)
+
+// ResourceGroupExists indicates whether a resource group exists within a subscription; otherwise false
+func ResourceGroupExists(t *testing.T, resourceGroupName string, subscriptionID string) bool {
+	result, err := ResourceGroupExistsE(resourceGroupName, subscriptionID)
+	require.NoError(t, err)
+	return result
+}
+
+// ResourceGroupExistsE indicates whether a resource group exists within a subscription
+// This function would fail the test if there is an error.
+func ResourceGroupExistsE(resourceGroupName, subscriptionID string) (bool, error) {
+	subscriptionID, err := getTargetAzureSubscription(subscriptionID)
+	if err != nil {
+		return false, err
+	}
+	client, err := GetResourceGroupClientE(subscriptionID)
+	if err != nil {
+		return false, err
+	}
+	rg, err := client.Get(context.Background(), resourceGroupName)
+	if err != nil {
+		return false, err
```

Comment:
This should check the error for the "not exists" case and return `nil` for the error. Otherwise, `ResourceGroupExists` will never return `false`. See https://github.com/gruntwork-io/terratest/blob/master/modules/azure/networkinterface.go#L20 where we do this.

Reply:
The E suffix function will attempt to get required resources, i.e., subscription, resource group client, etc. and if successfully compare the resource group name to the expected parameter input (T|F).  This is same pattern used in previous approved Exists functions.

Reply:
The expectation of a `ResourceGroupExists` function that returns `bool` is to return `false` if the resource doesn't exist, not to fail the test if it doesn't exist (if we want it to fail, then it should be called `AssertResourceGroupExists`).

Right now, as implemented, `ResourceGroupExists` will never return `false` because all the cases that return `false` in `ResourceGroupExistsE` returns it with a non-nil error (so the `require.NoError` call will fail the test). This breaks the expectation of the function.

So we should either:

- Add in the check to the error to only bubble it up if it is an error that doesn't indicate the resource doesn't exist.
- Rename the function to `AssertResourceGroupExists` and remove the `bool` return value.

> This is same pattern used in previous approved Exists functions.

We may have slipped through this on previous functions. I caught this in the PR referenced above. We should agree on what the right pattern is here, and fix it if it isn't the case in the merged code.

Reply:
We should have exist. @skunklab see https://github.com/gruntwork-io/terratest/blob/48e24df893c925adef331c342b7d3979da08f51c/modules/azure/availabilityset.go#L22.  This pattern should replace the nil check for line 30-32.

Reply:
modified